types/digitalMatter/g62/uplink.js

```diff
@@ -0,0 +1,100 @@
+function hexToBytes(hex) {
+  let bytes;
+  let c;
+  for (bytes = [], c = 0; c < hex.length; c += 2) {
+    bytes.push(parseInt(hex.substr(c, 2), 16));
+  }
+  return bytes;
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const { port } = event.data;
+  const bytes = hexToBytes(payload);
+  const data = {};
+
+  if (port === 1 || port === 2) {
+    const gps = {};
+    const digital = {};
+    const lifecycle = {};
+    switch (bytes[0] & 0x3) {
+      case 0:
+        gps.tripType = "NO_TRIP";
+        break;
+      case 1:
+        gps.tripType = "IGNITION";
+        break;
+      case 2:
+        gps.tripType = "MOVEMENT";
+        break;
+      case 3:
+        gps.tripType = "RUN";
+        break;
+      default:
+        break;
+    }
+    gps.latitude =
+      (bytes[0] & 0xf0) +
+      bytes[1] * 256 +
+      bytes[2] * 65536 +
+      bytes[3] * 16777216;
+    if (gps.latitude >= 0x80000000) {
+      gps.latitude -= 0x100000000;
+    }
+    gps.latitude /= 1e7;
+
+    gps.longitude =
+      (bytes[4] & 0xf0) +
+      bytes[5] * 256 +
+      bytes[6] * 65536 +
+      bytes[7] * 16777216;
+    if (gps.longitude >= 0x80000000) {
+      gps.longitude -= 0x100000000;
+    }
+    gps.longitude /= 1e7;
+
+    digital.extPower = (bytes[0] & 0x4) !== 0;
+    gps.gpsFixCurrent = (bytes[0] & 0x8) !== 0;
+    digital.ignition = (bytes[4] & 0x1) !== 0;
+    digital.digitalInput1 = (bytes[4] & 0x2) !== 0;
+    digital.digitalInput2 = (bytes[4] & 0x4) !== 0;
+    digital.digitalOutput = (bytes[4] & 0x8) !== 0;
+    gps.headingDeg = bytes[8] * 2;
+    gps.speedKmph = bytes[9];
+    lifecycle.voltage = parseFloat((bytes[10] * 0.02).toFixed(3));
+
+    if (port === 1) {
+      digital.extVoltage = parseFloat(
+        (0.001 * (bytes[11] + bytes[12] * 256)).toFixed(3),
```

Comment:
explain this calculation

Reply:
I don't have an explanation in the documentation for this '^^

Reply:
ok